quinn-proto/src/connection/mod.rs

```diff
@@ -477,13 +482,17 @@ impl Connection {
                     SpaceId::Data,
                     "PATH_CHALLENGE queued without 1-RTT keys"
                 );
-                let mut buf = BytesMut::with_capacity(self.path.current_mtu() as usize);
-                let buf_capacity = self.path.current_mtu() as usize;
+
+                if buf.capacity() < self.path.current_mtu() as usize {
```

Comment:
I don't think this check is right? Previously we were checking the capacity of a newly created `BytesMut`, but now we're getting a pre-existing `BytesMut` from another callers -- instead of the total capacity we should be making sure that enough capacity is remaining beyond the current contents.